modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +365,202 @@ public void testGetAllIdPs() throws Exception {
+            "page number")
+    public void testGetPaginatedIdPsInfoWithIncorrectPageNumber() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        int pageNumber = 2;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        if (idpList.size() > 0) {
+            Assert.fail("Invalid idp list found while calling getPaginatedIdPsInfo() with incorrect page number.");
+        }
+    }
+
+    @Test(priority = 12, groups = "wso2.is", description = "test getAllIdpCount operation")
+    public void testGetAllIdPsCount() throws Exception {
+
+        int idpCount = idpMgtServiceClient.getAllIdpCount();
+        log.info("All idp paginated filtered count : " + idpCount);
+        Assert.assertEquals(idpCount, 2, "Total Idp Count not found ");
+    }
+
+    @Test(priority = 13, groups = "wso2.is", description = "test getFilteredIdpCount operation with incorrect filter")
+    public void testGetFilteredIdpCountWIthIncorrectFilter() {
+
+        try {
+            String filter = "names sp \"Search\" and namees ew \"Test\"";
+            int idpCount = idpMgtServiceClient.getFilteredIdpCount(filter);
+            if (idpCount > 0) {
+                Assert.fail("Invalid idp count found while calling getFilteredIdpCount() with incorrect filter.");
+            }
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidFilterErrorMessage, "Error occurred while testing" +
+                    " testGetFilteredIdpCountWIthIncorrectFilter() with incorrect filter.");
+        }
+    }
+
+    @Test(priority = 14, groups = "wso2.is", description = "test getFilteredIdpCount operation with incorrect filter " +
+            "attribute value")
+    public void testGetFilteredIdpCountWithIncorrectFilterAttributeValue() throws Exception {
+
+        String filter = "name sw \"Admin\"";
+        int idpCount = idpMgtServiceClient.getFilteredIdpCount(filter);
+        log.info("Idp count with incorrect filter attribute value: " + idpCount);
+        if (idpCount > 0) {
+            Assert.fail(
+                    "Invalid idp count found while calling getFilteredIdpCount() with incorrect filter attribute value");
+        }
+    }
+
+    @Test(priority = 14, groups = "wso2.is", description = "test getFilteredIdpCount operation with empty filter")
+    public void testGetFilteredIdpCountWithEmptyFilterValue() throws Exception {
+
+        String filter = "";
+        int idpCount = idpMgtServiceClient.getFilteredIdpCount(filter);
+        log.info("Idp count with empty filter: " + idpCount);
+        if (idpCount != 2) {
+            Assert.fail("Invalid Idp count found while calling getFilteredIdpCount() with empty filter value ");
+        }
+    }
+
+    @Test(priority = 15, groups = "wso2.is", description = "test getFilteredIdpCount operation")
+    public void testGetAllFilteredIdPsCount() throws Exception {
```

Comment:
change name as testGetFilteredIdPCountAfterIDPDeletion or something that conveys the test functionality

Reply:
This method is for testing idp total count, I moved the delete operation into the atEnd() method